integration/batching-with-context/batching.ts

```diff
@@ -78,7 +78,9 @@ export const BatchQueryRequest = {
 
   fromJSON(object: any): BatchQueryRequest {
     const message = createBaseBatchQueryRequest();
-    message.ids = (object.ids ?? []).map((e: any) => String(e));
+    if (Array.isArray(object?.ids)) {
```

Comment:
I see, so the idea is to get rid of the `createBase...()` call?

It made me wonder already wondering why the default values for each field were overwritten each time. Thanks for clarifying, now it makes more sense.

I'll change back to the ternary style.

Reply:
> I see, so the idea is to get rid of the createBase...() call

Yep! @aikoven has an in-flight PR that is either doing this soon, or at least moving us closer to that I think.

Reply:
@stephenh Ah I didn't know @aikoven was already working on this. So I actually just made a PR #463 to do what you described. It was kind of simple as I was already deep into this part of the code, and had a solution in mind for the trickier 'oneof' case.

So, @aikoven, if you also had something in the works and want to finish that instead, please close my PR. Sorry if I got in the way.

Reply:
Thanks for adapting. Yeah the overall direction is to move away from mutability where you never know for sure what is set when. So a lot of recent work sent into exactly one assignment per field.

This PR nicely adds to that journey

Reply:
> I see, so the idea is to get rid of the createBase...() call?

We can easily get rid of it in fromJSON and fromPartial. For `decode` it will probably still be required as we are reading the serialized data into the object in unknown order and not all fields are serialized.

Reply:
Yeah, for `decode` I was thinking that calling `createBase()` to get an object literal with every field assigned (right now I don't think it does literally every field, just some primitives), and then have the `decode` for loop just assign over existing keys, would be most likely to trigger the supposedly / I'm-admittedly-cargo-culting-a-little-bit optimization of "create objects with all keys set and then don't add/remove keys". :-)

Reply:
> right now I don't think it does literally every field, just some primitives

That changed in #457, see e.g. https://github.com/stephenh/ts-proto/pull/457#discussion_r776266306. Now all fields are set to default and decode can override them.

Reply:
Ah great!